osquery/core/query.cpp

```diff
@@ -545,11 +569,38 @@ Status serializeQueryData(const QueryData& q,
   return Status();
 }
 
-bool addUniqueRowToQueryData(QueryData& q, const Row& r) {
+bool addUniqueRowToQueryData(QueryDataTyped& q, const RowTyped& r) {
   if (std::find(q.begin(), q.end(), r) != q.end()) {
     return false;
   }
   q.push_back(r);
   return true;
 }
+
+void escapeNonPrintableBytes(std::string& data) {
+  std::string escaped;
+  // clang-format off
+  char const hex_chars[16] = {
+    '0', '1', '2', '3', '4', '5', '6', '7',
+    '8', '9', 'A', 'B', 'C', 'D', 'E', 'F',
+  };
+  // clang-format on
+
+  bool needs_replacement = false;
+  for (size_t i = 0; i < data.length(); i++) {
+    if (((unsigned char)data[i]) < 0x20 || ((unsigned char)data[i]) >= 0x80) {
```

Comment:
Will http://www.cplusplus.com/reference/locale/isprint/ work for your use case?

Reply:
Mmmm, probably.   I moved this from sql.cpp in order to keep using it in SQLInternal with SQLInternal no longer a subclass of SQL.  If you feel that we should replace in this change, I will.  Otherwise we can defer looking at replacement to a separate change.

Reply:
Sure, I noticed it later. Feel free to leave out comments for the code you just moved. Will try to fix those later myself.